src/nvim/normal.c

```diff
@@ -7533,6 +7534,12 @@ static void nv_put_opt(cmdarg_T *cap, bool fix_indent)
         empty = (curbuf->b_ml.ml_flags & ML_EMPTY);
         msg_silent--;
 
+        if (save_unnamed) {
+          // Note: Discard the result
+          if (op_reg_set_previous('0')) {
```

Comment:
This doesn't work if the previous register is not register 0. This is not covered by tests.

Reply:
Well, but how to get the current previous name?
neovim does not have Vim's function.  So it is hard to port the function.

Reply:
I'm also still thinking about this.